src/link.cpp

```diff
@@ -791,6 +791,8 @@ static Buf *build_a_raw(CodeGen *parent_gen, const char *aname, Buf *full_path,
         new_link_lib->provided_explicitly = parent_gen->libc_link_lib->provided_explicitly;
     }
 
+    child_gen->function_sections = true;
```

Comment:
This was added in `build_a_raw`, because both libc.zig and compiler_rt.zig are being compiled through that function.